cmd/helm/helm.go

```diff
@@ -166,7 +166,28 @@ func markDeprecated(cmd *cobra.Command, notice string) *cobra.Command {
 	return cmd
 }
 
-func setupConnection(c *cobra.Command, args []string) error {
+func waitTillerPodAndSetupConnection(waitTimeout int) error {
+	_, client, err := getKubeClient(settings.KubeContext)
```

Comment:
I think my only nit is that we're grabbing a kube client here and again during `setupConnection()` but I can understand how this is unavoidable given the way things are. Let me know if you have any ideas for improving that in the future given you're now familiar with the code.

Reply:
@michelleN I propose the following solution, https://github.com/kubernetes/helm/pull/3506/commits/c4a01b7229708715a2d680142d744359bc70fc50.